src/utils/misc.ts

```diff
@@ -0,0 +1,9 @@
+type VERSIONS = "v2.0.0";
+
+export function deprecated(thing: string, version: VERSIONS, opts?: { replacement?: string }): void {
```

Comment:
Cache whether the notice has been written for this function before, otherwise it can spam the console.

I'd actually suggest making this into a decorator instead which makes the code "cleaner" - it also means you can replace the function call with the original after the first call so no need to cache it (hmu if you need to!)

I've opened a poll on #190 which will decide whether to do this style, or fix them and provide legacy endpoints (I'm leaning towards this personally, and it's the more "lz-string" way, but  if we get a strong opinion another way I'll happily go with it)

Reply:
I started to write a reply and go through the various ways it could be done (as we can't use a real typescript / javascript decorator outside of a class) - and realised that it's premature optimisation and we can worry about this later - plus it's far more useful for developers than for the end-user, so we've got until the final release of v2 to improve matters, with this as a good place to start from :-)